app/src/main/java/ch/epfl/sdp/geometry/Area.java

```diff
@@ -1,8 +1,124 @@
 package ch.epfl.sdp.geometry;
 
+import ch.epfl.sdp.map.Displayable;
+
 /**
  * Represents an area in the 2D plane.
  */
-public interface Area {
-    boolean isInside(Vector vector);
+public abstract class Area implements Positionable, Displayable {
+    GeoPoint center;
+    GeoPoint oldCenter;
+    GeoPoint newCenter;
+    double time;
+    double finalTime;
+    boolean isShrinking;
+
+    /**
+     * A constructor for an area
+     *
+     * @param center the center of the area
+     */
+    public Area(GeoPoint center) {
+        this.center = center;
+    }
+
+    /**
+     * This method tells if the area is currently shrinking
+     *
+     * @return a boolean that tells if the area is currently shrinking
+     */
+    public boolean isShrinking() {
+        return isShrinking;
+    }
+
+    /**
+     * This method tells if the given location is inside the area
+     *
+     * @param geoPoint the location we want to know if it is inside the area
+     * @return a boolean that tells if the given location is inside the area
+     */
+    public boolean isInside(GeoPoint geoPoint) {
+        Vector vec = center.toVector().subtract(geoPoint.toVector());
+        System.out.println(vec.x());
+        System.out.println(vec.y());
```

Comment:
Maybe those lines are not needed anymore.

Reply:
whoops